gcs/gcs.go

```diff
@@ -0,0 +1,363 @@
+// Copyright (c) 2016-2017 The btcsuite developers
+// Copyright (c) 2016-2017 The Lightning Network Developers
+// Copyright (c) 2018 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package gcs
+
+import (
+	"encoding/binary"
+	"errors"
+	"math"
+	"sort"
+
+	"github.com/aead/siphash"
+	"github.com/decred/dcrd/chaincfg/chainhash"
+	"github.com/kkdai/bstream" // TODO: can probably remove this dep
```

Comment:
Yes, let's address this TODO and remove this please.  I really would prefer to avoid yet another dependency for something this trivial.